File: 02-arrays/easy/05-moveZero.cpp
// You are given an array of integers, your task is to move all the zeros in the array to the end of
// the array and move non-negative integers to the front by maintaining their order.
#include <bits/stdc++.h>
using namespace std;
void zeroToLeft(vector<int> &arr) {
    int n = arr.size();
    int j = -1;
    int i = 0;
    for (int k = 0; k < n; k++) {
        if (arr[k] == 0) {
            j = k;
            break;
        }
    }
    if (j == -1)
        return;
    for (int i = j + 1; i < n; i++) {
        if (arr[i] != 0) {
            int t = arr[i];
            arr[i] = arr[j];
            arr[j] = t;
            j++;
        }
    }
}

File: 02-arrays/easy/05-moveZero.go
package main

import "fmt"

func MoveZero(arr []int) {
	n := len(arr)
	i := 0
	j := 0
	for k := 0; k < n; k++ {
		if arr[k] == 0 {
			j = k
			break
		}
	}

	if j == -1 {
		return
	}
	i = j + 1
	for i < n {
		if arr[i] != 0 {
			arr[i], arr[j] = arr[j], arr[i]
			j++
		}
		i++
	}
}

func main() {
	arr := []int{1, 0, 2, 3, 2, 0, 0, 4, 5, 1}
	MoveZero(arr)
	fmt.Println(arr)
}
